cmd/juju/model/destroy.go

```diff
@@ -287,16 +309,28 @@ func (c *destroyCommand) Run(ctx *cmd.Context) error {
 			maxWait = &zeroSec
 		}
 	}
-	modelTag := names.NewModelTag(modelDetails.ModelUUID)
+
 	var timeout *time.Duration
 	if c.timeout >= 0 {
 		timeout = &c.timeout
 	}
 	if err := api.DestroyModel(modelTag, destroyStorage, force, maxWait, timeout); err != nil {
-		return c.handleError(
-			modelTag, modelName, api,
-			errors.Annotate(err, "cannot destroy model"),
-		)
+		err = errors.Annotate(err, "cannot destroy model")
+
+		if params.IsCodeOperationBlocked(err) {
+			return block.ProcessBlockedError(err, block.BlockDestroy)
+		}
+		if params.IsCodeHasPersistentStorage(err) {
+			modelStatus, err := getModelStatus(modelTag, api)
```

Comment:
we have a second modelstatus query here because time has elapsed from the previous query

Reply:
yup